Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -394,6 +394,51 @@ public static void RunPostDeploymentScripts(TraceListener tracer)
             }
         }
 
+        /// <summary>
+        /// As long as the task was not completed, we will keep updating the marker file.
+        /// The routine completes when either task completes or timeout.
+        /// If task is completed, we will remove the marker.
+        /// If timeout, we will leave the stale marker.
+        /// </summary>
+        public static async Task TrackPendingOperation(Task task, TimeSpan timeout)
+        {
+            const int DefaultTimeoutMinutes = 30;
+            const int DefaultUpdateMarkerIntervalMS = 10000;
+            const string MarkerFilePath = @"%TEMP%\SCMPendingOperation.txt";
+
+            // only applicable to azure env
+            if (!Environment.IsAzureEnvironment())
+            {
+                return;
+            }
+
+            if (timeout <= TimeSpan.Zero || timeout >= TimeSpan.FromMinutes(DefaultTimeoutMinutes))
+            {
+                // track at most N mins by default
+                timeout = TimeSpan.FromMinutes(DefaultTimeoutMinutes);
+            }
+
+            var start = DateTime.UtcNow;
+            var markerFile = System.Environment.ExpandEnvironmentVariables(MarkerFilePath);
+            while (start.Add(timeout) >= DateTime.UtcNow)
```

Comment:
So if we go over that  time, you just stop updating the marker file, but that doesn't affect the ongoing operation, right? I guess it just means that at that point you're letting DWAS kill it? Though if feels a bit strange that in that case, you don't delete the file and just let it go stale.

Maybe I'm misreading this?

Reply:
Correct.   This PR has **no** effect on current behavior.   It compliments by adding pending marker file.

Reply:
As discussed, it's best to still clean up the file in that case

Reply:
will do